TLM/TLM/UI/Helpers/SerializableUIOptionBase.cs

```diff
@@ -9,34 +9,76 @@ namespace TrafficManager.UI.Helpers {
 
     public abstract class SerializableUIOptionBase<TVal, TUI> : ILegacySerializableOption
         where TUI : UIComponent {
-        //Data:
-        public SerializableUIOptionBase(string fieldName) {
+
+        /* Data: */
+        public SerializableUIOptionBase(string fieldName, Options.PersistTo scope) {
+
             ValueField = typeof(Options).GetField(fieldName, BindingFlags.Static | BindingFlags.Public);
             if (ValueField == null) {
-                throw new Exception($"{typeof(Options)}.{fieldName} does not exists");
+                throw new Exception($"{typeof(Options)}.{fieldName} does not exist");
             }
+
+            Scope = scope;
         }
+
+        public TranslatorDelegate Translator;
         private FieldInfo ValueField;
-        private Options OptionInstance => Singleton<Options>.instance;
+        public Options.PersistTo Scope { get; private set; }
+
+        /// <summary>Gets or sets the value of the field this option represents.</summary>
         public virtual TVal Value {
             get => (TVal)ValueField.GetValue(null);
             set => ValueField.SetValue(null, value);
         }
+
+        /// <summary>Custom translation delegate.</summary>
+        /// <param name="key">The locale key to translate.</param>
+        /// <returns>Returns the translation for <paramref name="key"/>.</returns>
+        public delegate string TranslatorDelegate(string key);
+
         public static implicit operator TVal(SerializableUIOptionBase<TVal, TUI> a) => a.Value;
+
+        // Legacy serialisation in `OptionsManager.cs`
         public abstract void Load(byte data);
         public abstract byte Save();
 
-        //UI:
-        public abstract void AddUI(UIHelperBase container);
-        protected TUI _ui;
+        // Unknown purpose.
+        private Options OptionInstance => Singleton<Options>.instance;
+
+        /* UI: */
         public string Label;
         public string Tooltip;
         public bool Indent = false;
+        protected TUI _ui;
+        public abstract void AddUI(UIHelperBase container);
+
+        public virtual bool ReadOnlyUI { get; set; }
+
+        /// <summary>
+        /// Returns <c>true</c> if user can change the setting in the current scope.
```

Comment:
maybe replace `scope` with `<see cref="Scope"/>` or add`<seealso cref="Scope"/>` at the end

Reply:
done 👍